src/v/v8_engine/environment.h

```diff
@@ -10,32 +10,70 @@
 
 #pragma once
 
+#include "v8_engine/executor.h"
+
 #include <libplatform/libplatform.h>
 
 #include <v8.h>
 
 namespace v8_engine {
 
-// This class contains environment for v8.
-// Can be used only one time per application.
-class enviroment {
+class v8_platform_wrapper {
 public:
-    // Init single thread v8::Platform. If we use NewDefaultPlatform platform
-    // it will create threadpool for background tasks.
-    // Environment must be created before all v8 instance are starting!
-    enviroment();
+    // Init single thread v8::Platform. If we use
+    // NewDefaultPlatform platform it will create threadpool for background
+    // tasks. Environment must be created before all v8 instance are starting!
+    v8_platform_wrapper();
 
-    enviroment(const enviroment& other) = delete;
-    enviroment& operator=(const enviroment& other) = delete;
-    enviroment(enviroment&& other) = default;
-    enviroment& operator=(enviroment&& other) = default;
+    v8_platform_wrapper(const v8_platform_wrapper& other) = delete;
+    v8_platform_wrapper& operator=(const v8_platform_wrapper& other) = delete;
+    v8_platform_wrapper(v8_platform_wrapper&& other) = delete;
+    v8_platform_wrapper& operator=(v8_platform_wrapper&& other) = delete;
 
-    // Stop v8 envoroment and release any resources used by v8.
+    // Stop v8 platform and release any resources used by v8.
     // Must be called after all v8 instances are finished!
-    ~enviroment();
+    ~v8_platform_wrapper();
 
 private:
     std::unique_ptr<v8::Platform> _platform;
 };
 
+class executor_wrapper {
```

Comment:
I do not understand the need for this `executor_wrapper`. What is the idea behind it ?

Reply:
Maybe it would be better to leave wrapper implemention until we will want to built executor in redpanda.

Reply:
delete it